packages/babel-core/src/transformation/file/options/build-config-chain.js

```diff
@@ -60,15 +65,26 @@ class ConfigChainBuilder {
     while (loc !== (loc = path.dirname(loc))) {
       if (!foundConfig) {
         const configLoc = path.join(loc, BABELRC_FILENAME);
-        if (exists(configLoc)) {
-          this.addConfig(configLoc);
-          foundConfig = true;
-        }
-
+        const configJSLoc = path.join(loc, BABELRC_JS_FILENAME);
         const pkgLoc = path.join(loc, PACKAGE_FILENAME);
-        if (!foundConfig && exists(pkgLoc)) {
-          foundConfig = this.addConfig(pkgLoc, "babel", JSON);
-        }
+        const configLocs = [configLoc, configJSLoc, pkgLoc];
+        const foundConfigs = configLocs.reduce((arr, config) => {
+          if (exists(config)) {
+            const configAdded = config === pkgLoc
+              ? this.addConfig(config, "babel", JSON)
+              : this.addConfig(config);
+
+            if (configAdded && arr.length) {
```

Comment:
I'm concerned that this could be a breaking change in 6.x, since it could be pretty easy for someone first starting with Babel to have creating both a `.babelrc` and an entry in `package.json`, and this would break them, if I'm reading it right.

Could we error if `.babelrc.js` is present with one of the other two?